lib/cipherstash/index/dynamic_filter_match.rb

```diff
@@ -0,0 +1,45 @@
+module CipherStash
+    class Index
+      # Implementation of the dynamic-filter-match index
+      #
+      # @private
+      class DynamicFilterMatch < Index
+        INDEX_OPS = {
+          "match" => -> (idx, s) do
+            filter = BloomFilter.new([meta_settings["$prfKey"]].pack("H*"))
+
+            bits = idx.text_processor.perform(s)
+              .map { |t| ore_encrypt(t).to_s }
+              .reduce(filter) { |filter, term| filter.add(term) }
+              .bits # TODO: should bits be a list rather than set?
```

Comment:
They're both enumerables, so I guess as long as the protobuf encoder isn't outrageously picky, it'll presumably work either way.